pkg/nodeconfig/nodeconfig.go

```diff
@@ -288,6 +304,186 @@ func (nc *nodeConfig) Configure() error {
 	return err
 }
 
+// CreateBootstrapFiles creates all prerequisite files on the node required to start kubelet using latest ignition spec
+func (nc *nodeConfig) createBootstrapFiles() error {
+	if err := nc.createFilesFromIgnition(); err != nil {
+		return err
+	}
+
+	if err := nc.createBootstrapKubeconfig(); err != nil {
+		return err
+	}
+
+	clusterDNS, err := cluster.GetDNS(nc.clusterServiceCIDR)
+	if err != nil {
+		return err
+	}
+	if err := nc.createKubeletConf(clusterDNS); err != nil {
+		return err
+	}
+	return nil
+}
+
+// createFilesFromIgnition creates any files it can from ignition: kubelet CA cert, cloud-config file
+func (nc *nodeConfig) createFilesFromIgnition() error {
+	// For each new file in the ignition file check if is a file we are interested in, if so, decode it
+	// and write the contents to a temporary destination path
+	filesToTranslate := map[string]struct{}{
+		ignition.KubeletCACertPath: {},
+	}
+
+	ign, err := ignition.New(nc.client)
+	if err != nil {
+		return err
+	}
+	kubeletArgs, err := ign.GetKubeletArgs()
+	if err != nil {
+		return err
+	}
+	if _, ok := kubeletArgs[ignition.CloudConfigOption]; ok {
+		filesToTranslate[ignition.CloudConfigPath] = struct{}{}
+	}
+
+	for _, ignFile := range ign.Config.Storage.Files {
+		if _, ok := filesToTranslate[ignFile.Node.Path]; ok {
+			if ignFile.Contents.Source == nil {
+				return errors.Errorf("could not process %s: File is empty", ignFile.Node.Path)
+			}
+			contents, err := dataurl.DecodeString(*ignFile.Contents.Source)
+			if err != nil {
+				return errors.Wrapf(err, "could not decode %s", ignFile.Node.Path)
+			}
+
+			fileName := filepath.Base(ignFile.Node.Path)
+			if err := nc.Windows.EnsureFileContent(contents.Data, fileName, windows.K8sDir); err != nil {
+				return err
+			}
+			nc.log.V(1).Info("created", "file", windows.K8sDir+fileName)
+		}
+	}
+	return nil
+}
+
+// createBootstrapKubeconfig creates a kubeconfig for kubelet to initially communicate with the API server
+func (nc *nodeConfig) createBootstrapKubeconfig() error {
+	nodeBootstrapToken, err := nc.k8sclientset.CoreV1().Secrets(mcoNamespace).Get(context.TODO(), mcoBootstrapSecret,
+		meta.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	kubeconfig, err := generateKubeconfig(nodeBootstrapToken, nodeConfigCache.apiServerEndpoint)
+	if err != nil {
+		return err
+	}
+	kubeconfigData, err := json.Marshal(kubeconfig)
+	if err != nil {
+		return err
+	}
+	kubeconfigFileName := "bootstrap-kubeconfig"
+	if err := nc.Windows.EnsureFileContent(kubeconfigData, kubeconfigFileName, windows.K8sDir); err != nil {
+		return err
+	}
```

Comment:
Please use filepath.Base and filepath.Dir to get the arguments for EnsureFileContent instead of `kubeconfigFileName`

Reply:
Having trouble with this suggestion. Since WMCO runs as a linux process, `filepath` is not handling Windows paths properly.
```golang
func test() {
	path := "C:\\k\\bootstrap-kubeconfig"
	fileName := filepath.Base(path)
	dir := filepath.Dir(path)

	fmt.Println("Dir:", dir)
	fmt.Println("File:", fileName)
}
```
Output: 
```
Dir: .
File: C:\k\bootstrap-kubeconfig
```

Reply:
Would [FromSlash](https://cs.opensource.google/go/go/+/go1.19.2:src/path/filepath/path.go;l=188) [¶](https://pkg.go.dev/path/filepath#FromSlash) or [ToSlash](https://cs.opensource.google/go/go/+/go1.19.2:src/path/filepath/path.go;l=178) [¶](https://pkg.go.dev/path/filepath#ToSlash) help?

Reply:
Nope, the separator in the `filepath` library is based on the OS. Since WMCO does not run on Windows, it can't handle Windows paths.

Reply:
I can do some string manipulation here instead. Find last occurrence of `\` and split string at that index to get the dir and fileName